From 94c7219902d2be47b0827bbf8d217fe3b01db819 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Wed, 12 Feb 2020 05:41:03 -0700
Subject: [PATCH 1183/1852] mtd: spi-nor: update ptr pointer after
 spi_nor_read_data

commit 063d6f3ca5c787baf7e7bc039878971301b6fefb from
https://github.com/Xilinx/linux-xlnx.git

In spi_nor_spimem_xfer_data, it will limit the transfer size
according to nor->bouncebuf_size which is usually equal to
PAGE_SIZE. This means that sometimes a spi_nor_read operation
will be split into several spi_nor_read_data operations if
the transfer size is bigger than PAGE_SIZE .

Considering the case above, when is_ofst_odd is 0, we initialize
the "ptr" to be the "buf" and pass "ptr" as the argument to
spi_nor_read_data, the "ptr" should be updated as the "buf"
after spi_nor_read_data is called, or else the receive data
will be stored in the same place every time.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 37d90d8c987a..b87f94f54165 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2848,6 +2848,8 @@ static int spi_nor_read(struct mtd_info *mtd, loff_t from, size_t len,
 			*retlen += ret;
 		}
 		buf += ret;
+		if (!is_ofst_odd)
+			ptr += ret;
 		from += ret;
 		len -= ret;
 	}
-- 
2.31.1

